File: src/main/java/spring/webmvc/domain/model/entity/Permission.java
package spring.webmvc.domain.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Permission extends BaseTime {

	@Id
	@GeneratedValue
	@Column(name = "permission_id")
	private Long id;
	private String name;
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/Permission.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id

@Entity
class Permission protected constructor(
    val name: String,
) : BaseTime() {
    @Id
    @GeneratedValue
    @Column(name = "permission_id")
    var id: Long? = null
        protected set
}